agreementbot/agreementworker.go

```diff
@@ -651,8 +638,12 @@ func (b *BaseAgreementWorker) InitiateNewAgreement(cph ConsumerProtocolHandler,
 		} else {
 
 			foundWorkload = true
-			svcIds = make([]string, len(sIds))
-			copy(svcIds, sIds)
+
+			// copy all the service ids
```

Comment:
Can you use the built in copy function here?

Reply:
The depServices is a map, we just need its keys in svcIds.